### PR TITLE
Support describe for views

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
This pr:

1) Introduces a new analyze rules to ensure `describe` works on views
2) Fixes a bug in the return of SHOW COLUMNS
3) Makes sure describe works with expressions

### CODE REVIEW

File: enginetest/queries/script_queries.go [hunk 1]
```diff
@@ -2013,6 +2013,21 @@ var ScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "Describe with expression correctly works",
```
Comment (VinaiRachakonda) [inline enginetest/queries/script_queries.go:2017]: Can add additional view tests here. Will add after review

File: enginetest/queries/script_queries.go [hunk 2]
```diff
@@ -2013,6 +2013,21 @@ var ScriptTests = []ScriptTest{
...
+		SetUpScript: []string{
+			"CREATE TABLE t(pk int primary key, val int DEFAULT (pk * 2))",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "DESCRIBE t",
+				Expected: []sql.Row{
+					{"pk", "int", "NO", "PRI", "NULL", ""},
+					{"val", "int", "YES", "", "((pk * 2))", ""},
```
Comment (zachmu) [inline enginetest/queries/script_queries.go:2026]: This doesn't match the MySQL behavior (extra parens on  the default expr)

Comment (zachmu) [reply enginetest/queries/script_queries.go:2026]: MySQL itself is not consistent about this. Just file an issue for now, we'll want to fix it at some point.

```
mysql> desc t;
+-------+------+------+-----+------------+-------------------+
| Field | Type | Null | Key | Default    | Extra             |
+-------+------+------+-----+------------+-------------------+
| pk    | int  | NO   | PRI | NULL       |                   |
| val   | int  | YES  |     | (`pk` * 2) | DEFAULT_GENERATED |
+-------+------+------+-----+------------+-------------------+
2 rows in set (0.02 sec)

mysql> show create table t;
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table
                                                    |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `pk` int NOT NULL,
  `val` int DEFAULT ((`pk` * 2)),
  PRIMARY KEY (`pk`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

File: sql/analyzer/resolve_subqueries.go [hunk 3]
```diff
@@ -316,3 +316,45 @@ func setJoinScopeLen(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, se
 		return n, transform.SameTree, nil
 	})
 }
+
+// setViewTargetSchema is used to set the target schema for views. It is run after resolve_subqueries in order for
+// SubqueryAlias resolution to happen.
+func setViewTargetSchema(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
+	span, ctx := ctx.Span("set_view_target_schema")
```
Comment (zachmu) [inline sql/analyzer/resolve_subqueries.go:323]: Restrict this to just the specific node or nodes you are targeting here

**Review by zachmu — APPROVED** [review]
LGTM just a couple small comments

File: enginetest/queries/script_queries.go (hunk 2 above)
Comment (VinaiRachakonda) [reply enginetest/queries/script_queries.go:2026]: Sounds good

**Pull request merged**